tests/foreman/cli/test_classparameters.py

```diff
@@ -123,27 +132,50 @@ def setUpClass(cls):
         """Import some parametrized puppet classes. This is required to make
         sure that we have smart class variable available.
         Read all available smart class parameters for imported puppet class to
-        be able to work with unique entity for each specific test. Raise an
-        exception and skip all tests in case not enough parameters returned.
+        be able to work with unique entity for each specific test.
         """
         super(SmartClassParametersTestCase, cls).setUpClass()
-        cls.host_name = settings.server.hostname
-        ssh.command('puppet module install --force puppetlabs/ntp')
-        cls.env = Environment.info({u'name': 'production'})
-        Proxy.importclasses({
+        cls.puppet_module = "robottelo/cli_test_classparameters"
+        cls.org = make_org()
+        # Create product and puppet modules repository
+        product = make_product({u'organization-id': cls.org['id']})
+        repo = make_repository({
+            u'product-id': product['id'],
+            u'content-type': 'puppet',
+            u'url': FAKE_8_PUPPET_REPO,
+        })
+        # Synchronize repo via provided URL
+        Repository.synchronize({'id': repo['id']})
+        # Add selected module to Content View
+        cv = make_content_view({u'organization-id': cls.org['id']})
+        ContentView.puppet_module_add({
+            u'author': cls.puppet_module.split('/')[0],
+            u'name': cls.puppet_module.split('/')[1],
+            u'content-view-id': cv['id'],
+        })
+        # CV publishing will automatically create Environment and will make
+        # puppet classes from module available
+        ContentView.publish({u'id': cv['id']})
+        cls.env = Environment.list({
+            u'search': 'content_view="{0}"'.format(cv['name'])
+        })[0]
+        cls.puppet = Puppet.info({
+            u'name': cls.puppet_module.split('/')[1]
+        })
+        cls.sc_params_list = SmartClassParameter.list({
             u'environment': cls.env['name'],
-            u'name': cls.host_name,
+            u'search': 'puppetclass="{0}"'.format(cls.puppet['name'])
         })
-        cls.puppet = Puppet.info({u'name': 'ntp'})
-        sc_params_list = SmartClassParameter.list({
-            'environment': cls.env['name'],
-            'search': 'puppetclass=ntp'
-        })
-        if len(sc_params_list) < 45:
-            raise RuntimeError('There are not enough smart class parameters to'
-                               ' work with in provided puppet class')
         cls.sc_params_ids_list = [
-            sc_param['id'] for sc_param in sc_params_list]
+            sc_param['id'] for sc_param in cls.sc_params_list]
+
+    def setUp(self):
+        """Checks that there is at least one not overridden
+        smart class parameter before executing test.
+        """
+        if len(self.sc_params_list) == 0:
+            raise Exception("Not enough smart class parameters. Please"
```

Comment:
nitpick: missing space after word `Please`. Your string will result in `Pleaseupdate puppet module.`